src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -851,7 +859,10 @@ public bool TakeMessageHeaders(SocketInput input, FrameRequestHeaders requestHea
                 while (!scan.IsEnd)
                 {
                     var beginName = scan;
-                    scan.Seek(ref _vectorColons, ref _vectorCRs);
+                    if (scan.Seek(ref _vectorColons, ref _vectorCRs) == -1)
+                    {
+                        return false;
```

Comment:
Looks like MessageBody is ok https://github.com/aspnet/KestrelHttpServer/blob/dev/src/Microsoft.AspNetCore.Server.Kestrel/Http/MessageBody.cs#L394-L399